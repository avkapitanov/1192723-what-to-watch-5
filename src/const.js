export const DEFAULT_RATING_FORM_VALUE = 3;
export const MAX_RATING_VALUE_FORM = 5;
export const HOVER_TIME_BEFORE_PLAYING = 2000;
export const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export const DEFAULT_ACTIVE_TAB = FilmTab.OVERVIEW;
export const DEFAULT_GENRE_FILTER_VALUE = `All genres`;
export const INITIAL_FILMS_COUNT = 8;
export const FILMS_COUNT_PER_STEP = 8;
export const MAX_GENRES_IN_FILTER = 9;
export const API_ENDPOINT = `https://5.react.pages.academy/wtw`;
export const REQUEST_TIMEOUT = 5000;
export const HttpResponseCode = {
  UNAUTHORIZED: 401
};
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
export const FilmRating = {
  BAD: {
    VALUE: 3,
    TEXT: `Bad`
  },
  NORMAL: {
    VALUE: 5,
    TEXT: `Normal`
  },
  GOOD: {
    VALUE: 8,
    TEXT: `Good`
  },
  VERY_GOOD: {
    VALUE: 10,
    TEXT: `Very good`
  },
  AWESOME: {
    TEXT: `Awesome`
  }
};

