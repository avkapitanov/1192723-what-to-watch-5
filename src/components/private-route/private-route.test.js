import React from "react";
import renderer from "react-test-renderer";
import PrivateRoute from "./private-route";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../browser-history";
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";
import {AuthorizationStatus} from "../../const";

const mockStore = configureStore([]);
const store = mockStore({
  USER: {
    authorizationStatus: AuthorizationStatus.AUTH,
  }
});

it(`PrivateRoute component render correctly`, () => {
  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter history={browserHistory}>
          <PrivateRoute
            exact={true}
            path={`/myfilms`}
            render={() => {}}
          />
        </BrowserRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
