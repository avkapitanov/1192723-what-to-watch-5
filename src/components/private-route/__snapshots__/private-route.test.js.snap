// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component render correctly 1`] = `null`;
